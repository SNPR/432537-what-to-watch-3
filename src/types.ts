export type Movie = {
  name: string;
  posterUrl: string;
  previewUrl: string;
  bigPosterUrl: string;
  backgroundColor: string;
  description: string;
  rating: number;
  votes: number;
  director: string;
  starring: string[];
  runTime: string;
  genre: string;
  releaseYear: number;
  id: number | string;
  isFavorite: boolean;
  videoUrl: string;
  trailerUrl: string;
};
