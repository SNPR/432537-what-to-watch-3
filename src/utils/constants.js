export const ALL_GENRES = `All genres`;

export const AppRoute = {
  LOGIN: `/login`,
  ROOT: `/`,
  ADD_REVIEW: `/add-review`,
  MY_LIST: `/mylist`,
  FILMS: `/films`,
  PLAYER: `/player`
};

export const SHOWED_MOVIES_DEFAULT = 8;

export const MIN_REVIEW_LENGTH = 50;
export const MAX_REVIEW_LENGTH = 400;

export const Tab = {
  Name: {
    OVERVIEW: `overview`,
    DETAILS: `details`,
    REVIEWS: `reviews`
  },
  ACTIVE_CLASS: `movie-nav__item--active`
};
