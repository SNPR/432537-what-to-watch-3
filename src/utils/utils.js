export const getTextRating = (value) => {
  if (value >= 0 && value < 3) {
    return `Bad`;
  } else if (value >= 3 && value < 5) {
    return `Normal`;
  } else if (value >= 5 && value < 8) {
    return `Good`;
  } else if (value >= 8 && value <= 10) {
    return `Very good`;
  }

  return ``;
};
