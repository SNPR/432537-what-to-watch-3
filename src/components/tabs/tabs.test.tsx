import * as React from "react";
import * as renderer from "react-test-renderer";
import Tabs from "./tabs";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import Namespace from "../../reducer/namespace";
import {Tab} from "../../utils/constants";
import {noop} from "../../utils/utils";
import {Movie} from "../../types";

const movie: Movie = {
  name: `MovieName`,
  posterUrl: `https://url.com`,
  previewUrl: `https://url.com`,
  bigPosterUrl: `https://url.com`,
  backgroundColor: `blue`,
  description: `Descrtiption`,
  rating: 9,
  votes: 3452,
  director: `Director`,
  starring: [`Artist 1`, `Artist 2`],
  runTime: `133`,
  genre: `Action`,
  releaseYear: 1995,
  id: 1,
  isFavorite: false,
  videoUrl: `https://url.com`,
  trailerUrl: `https://url.com`
};

it(`Should render Tabs component`, () => {
  const mockStore = configureStore([]);

  const store = mockStore({
    [Namespace.DATA]: {
      promoFilm: movie
    }
  });

  const tree = renderer
    .create(
        <Provider store={store}>
          <Tabs
            movie={movie}
            getActiveClass={noop}
            setActiveTab={noop}
            selectedTab={Tab.Name.DETAILS}
          />
        </Provider>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
