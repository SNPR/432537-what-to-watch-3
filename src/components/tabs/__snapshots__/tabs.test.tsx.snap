// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Tabs component 1`] = `
<div
  className="movie-card__desc"
>
  <nav
    className="movie-nav movie-card__nav"
  >
    <ul
      className="movie-nav__list"
    >
      <li
        className="movie-nav__item undefined"
      >
        <a
          className="movie-nav__link"
          onClick={[Function]}
        >
          Overview
        </a>
      </li>
      <li
        className="movie-nav__item undefined"
      >
        <a
          className="movie-nav__link"
          onClick={[Function]}
        >
          Details
        </a>
      </li>
      <li
        className="movie-nav__item undefined"
      >
        <a
          className="movie-nav__link"
          onClick={[Function]}
        >
          Reviews
        </a>
      </li>
    </ul>
  </nav>
  <div
    className="movie-card__text movie-card__row"
  >
    <div
      className="movie-card__text-col"
    >
      <p
        className="movie-card__details-item"
      >
        <strong
          className="movie-card__details-name"
        >
          Director
        </strong>
        <span
          className="movie-card__details-value"
        >
          Director
        </span>
      </p>
      <p
        className="movie-card__details-item"
      >
        <strong
          className="movie-card__details-name"
        >
          Starring
        </strong>
        <span
          className="movie-card__details-value"
        >
          Artist 1
           
          <br />
          Artist 2
           
          <br />
        </span>
      </p>
    </div>
    <div
      className="movie-card__text-col"
    >
      <p
        className="movie-card__details-item"
      >
        <strong
          className="movie-card__details-name"
        >
          Run Time
        </strong>
        <span
          className="movie-card__details-value"
        >
          2h 13m
        </span>
      </p>
      <p
        className="movie-card__details-item"
      >
        <strong
          className="movie-card__details-name"
        >
          Genre
        </strong>
        <span
          className="movie-card__details-value"
        >
          Action
        </span>
      </p>
      <p
        className="movie-card__details-item"
      >
        <strong
          className="movie-card__details-name"
        >
          Released
        </strong>
        <span
          className="movie-card__details-value"
        >
          1995
        </span>
      </p>
    </div>
  </div>
</div>
`;
